### ISSUE TITLE
`require-passing-this` rule is incorrectly applied to the `App` class.

### ISSUE THREAD

**Issue by sakurai-ryo** [ren-yamanashi/eslint-cdk-plugin#93]
### Describe the bug

You can only pass `Props` as arguments to the `App` class in `aws-cdk-lib`, but `require-passing-this` rule raises an error if `this` is not passed as an argument.

```ts
export declare class App extends Stage {
    constructor(props?: AppProps);
}
```

## Code example
```ts
import { App } from "aws-cdk-lib";
new App({});
```

## Flat config
```js
import eslintCdkPlugin from "eslint-cdk-plugin";
import tsEslint from "typescript-eslint";

export default [
  ...tsEslint.configs.recommended,
  {
    files: ["lib/**/*.ts", "bin/*.ts"],
    languageOptions: {
      parserOptions: {
        projectService: true,
        project: "./tsconfig.json",
      },
    },
    plugins: {
      cdk: eslintCdkPlugin,
    },
    rules: {
      ...eslintCdkPlugin.configs.recommended.rules,
    },
  },
];
```

## Repro  steps
Run the following command.
```bash
$ npx eslint@v9 --config eslint.config.mjs
```

### Expected Behavior

No error.

### Current Behavior

If you do not pass `this` to the `App` class, an error will occur

### eslint-cdk-plugin Version

1.0.3

### Node.js Version

20

### Language

TypeScript

### Language Version

5.6.3

### Additional Context

I think this is the same cause as the following issue.
https://github.com/ren-yamanashi/eslint-cdk-plugin/issues/78

### Checklist

- [x] I have searched for similar issues before creating this one
- [x] I have provided all the necessary information to reproduce the issue
- [x] I have verified this issue is related to eslint-cdk-plugin and not another dependency

**Comment by ren-yamanashi** [comment]
@sakurai-ryo 

Thanks for the bug report 🙏.  
I will fix it soon.
